paddle/fluid/pybind/generator_py.cc

```diff
@@ -39,7 +39,36 @@ void BindGenerator(py::module* m_ptr) {
       .def("current_seed",
            [](std::shared_ptr<phi::Generator::GeneratorState>& self) {
              return self->current_seed;
-           });
+           })
+#if defined(PADDLE_WITH_CUDA) || defined(PADDLE_WITH_HIP)
+      // NOTE(shenliang03): Due to the inability to serialize mt19937_64
+      // type, resulting in a problem with precision under the cpu.
+      .def(py::pickle(
+          [](const phi::Generator::GeneratorState& s) {  // __getstate__
+            return py::make_tuple(s.device, s.current_seed, s.thread_offset);
+          },
+          [](py::tuple s) {  // __setstate__
+            if (s.size() != 3)
+              throw std::runtime_error("Invalid Random state!");
```

Comment:
`Invalid Random state. Please check the format(device, current_seed, thread_offset)`.